h2/src/main/org/h2/jdbc/JdbcStatement.java

```diff
@@ -801,17 +801,29 @@ public int[] executeBatch() throws SQLException {
             debugCodeCall("executeBatch");
             checkClosed();
             if (batchCommands == null) {
-                batchCommands = new ArrayList<>();
+                closeOldResultSet();
```

Comment:
Are you sure `closeOldResultSet();` this shouldn't be called always? As we do in other methods.